app/src/main/java/org/groundplatform/android/MainViewModel.kt

```diff
@@ -126,6 +142,18 @@ constructor(
       userRepository.saveUserDetails(user)
       if (!isTosAccepted()) {
         MainUiState.TosNotAccepted
+      } else if (isDeepLinkAvailable()) {
+        val deepLinkUri = _deepLinkUri.value
+        val surveyId =
+          deepLinkUri
+            ?.takeIf { it.pathSegments.firstOrNull() == SURVEY_PATH_SEGMENT }
```

Comment:
Let's move the logic of parsing survey id from uri to a separate method for readability